pkg/util/hlc/hlc.go

```diff
@@ -251,3 +341,28 @@ func (c *Clock) UpdateAndCheckMaxOffset(rt Timestamp) (Timestamp, error) {
 	defer c.mu.Unlock()
 	return c.updateLocked(rt, false)
 }
+
+// lastPhysicalTime returns the last physical time
+func (c *Clock) lastPhysicalTime() int64 {
+	c.mu.Lock()
```

Comment:
have you checked that all existing reads/writes of this variable are under a lock?